src/components/notifications/NotificationsMenu.js

```diff
@@ -149,6 +133,57 @@ function NotificationsMenu(props) {
         },
     });
 
+    const { isMarked } = useQuery({
+        queryKey: NOTIFICATIONS_MARK_ALL_SEEN_KEY,
+        queryFn: markAllSeen,
+        config: {
+            onSuccess: () => {
+                setAllNotificationsSeen();
+            },
+            onError: (error) => {
+                console.log("Error marking all notifications as Seen");
+                // *** Will add this after understanding HOC better. ***
+                // showErrorAnnouncer(
+                //     t("errorMarkAsSeen", {
+                //         count: notifications.length,
+                //     }),
+                //     error
+                // );
+            },
+        },
+    });
+
+    const setAllNotificationsSeen = () => {
+        console.log(isMarked);
+        setNotifications(
+            notifications?.map((item) => {
+                const tempItem = item;
+                tempItem.seen = true;
+                return tempItem;
```

Comment:
It looks like `tempItem` is meant to be a copy, and this isn't quite right.  I'm sure in your CS classes you've come across or will very soon come across the idea of copying by value or copying by reference.  Javascript will copy an object, like `item` in this case, by reference.  

[Here](https://attacomsian.com/blog/javascript-clone-objects) is an article I found real quick that will explain why this is incorrect and some methods for doing it correctly.  We prefer the `spread` method, which is mentioned in that article, and I would suggest you use that here. Understanding destructuring and the spread syntax will be key in understanding a lot of our code base.